### PR TITLE
feat: add appendix b, gherkin suite

### PR DESCRIPTION
**Pull request by toddbaert**
Adds a new appendix for gherkin suites, taken from https://github.com/open-feature/test-harness. This is especially useful/relevant now that implementations are including their own in-memory providers. Implementations can use this suite + their in-memory provider for a robust suite of e2e tests.

Eventually, I'd like to add a "events" test suite too, but for now I'm only including the `evaluation.feature`.

### CODE REVIEW

File: specification/README.md [hunk 1]
```diff
@@ -16,7 +16,8 @@ sidebar_position: 0
 - [Evaluation Context](./sections/03-evaluation-context.md)
 - [Hooks](./sections/04-hooks.md)
 - [Events](./sections/05-events.md)
-- [Appendix-A](./appendix-A.md)
+- [Appendix A: Included Utilities](./appendix-a-included-utilities.md)
```
[the remark this answers is not in the document]
Comment (Kavindu-Dodan) [reply specification/README.md:19]: Thanks for the change

File: specification/appendix-b-gherkin-suites.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+title: Appendix A: End-to-End Tests
```
Comment (Kavindu-Dodan) [inline specification/appendix-b-gherkin-suites.md:2]:
```suggestion
title: Appendix B: End-to-End Tests
```

**Review by Kavindu-Dodan — APPROVED** [review]
LGTM

**Review by liran2000 — APPROVED** [review]

**Review by lukas-reining — APPROVED** [review]
Like that!

**Pull request merged**